gulpfile.js

```diff
@@ -173,67 +185,56 @@ gulp.task('create-sha256sum-of-exe', function(cb) {
   createSHA256File(installerExe, cb);
 });
 
-gulp.task('package', function(cb) {
+gulp.task('package', function(cb) {
   runSequence('create-7zip-archive', 'update-metadata', 'create-final-exe', 'create-sha256sum-of-exe', cb);
 });
 
 // for a given filename, return the sha256sum
-// eg., a7da255b161c2c648e8465b183e2483a3bcc64ea1aa9cbdc39d00eeb51cbcf38
-// getSHA256("C:\\Program Files\\Internet Explorer\\iexplore.exe", function(hashstring) { console.log("Got sha256 = "+ hashstring); } );
 function getSHA256(filename, cb) {
   var hashstring = "NONE";
   var hash = crypto.createHash('sha256');
-  //console.log("Generate SHA256 for " + filename);
   var readStream = fs.createReadStream(filename);
-  readStream
-    .on('readable', function () {
-      var chunk;
-      while (null !== (chunk = readStream.read())) {
-        hash.update(chunk);
-      }
-    })
-    .on('end', function () {
-      // a7da255b161c2c648e8465b183e2483a3bcc64ea1aa9cbdc39d00eeb51cbcf38
-      hashstring = hash.digest('hex');
-      // console.log("[1] SHA256 = " + hashstring);
-      cb(hashstring);
-    });
-  return hashstring;
+  readStream.on('readable', function () {
+    var chunk;
+    while (null !== (chunk = readStream.read())) {
+      hash.update(chunk);
+    }
+  }).on('end', function () {
+    hashstring = hash.digest('hex');
+    cb(hashstring);
+  });
 }
 
 // writes to {filename}.sha256, eg., 6441cde1821c93342e54474559dc6ff96d40baf39825a8cf57b9aad264093335 requirements.json
 function createSHA256File(filename, cb) {
+  !cb && cb();

```

Comment:
I'll remove that, it just force failure before calculating sha if cb is not defined.